tools/deployment/linux_post_install.sh

```diff
@@ -0,0 +1,45 @@
+#!/usr/bin/env bash
+
+#  Copyright (c) 2014-present, Facebook, Inc.
+#  All rights reserved.
+#
+#  This source code is licensed under the BSD-style license found in the
+#  LICENSE file in the root directory of this source tree. An additional grant
+#  of patent rights can be found in the PATENTS file in the same directory.
+
+SCRIPT_DIR="/usr/share/osquery/scripts"
+
+INITD_SRC="$SCRIPT_DIR/osqueryd.initd"
+INITD_DST="/etc/rc.d/init.d/osqueryd"
+INITD_DST_DEB="/etc/init.d/osqueryd"
+
+SYSTEMD_SERVICE_SRC="$SCRIPT_DIR/osqueryd.service"
+SYSTEMD_SERVICE_DST="/usr/lib/systemd/system/osqueryd.service"
+
+SYSTEMD_SYSCONFIG_SRC="$SCRIPT_DIR/osqueryd.sysconfig"
+SYSTEMD_SYSCONFIG_DST="/etc/sysconfig/osqueryd"
+SYSTEMD_SYSCONFIG_DST_DEB="/etc/default/osqueryd"
+
+# Check if OS is Debian based
+if ls /etc | grep -Fq debian_version; then
+  INITD_DST=$INITD_DST_DEB
+  SYSTEMD_SYSCONFIG_DST=$SYSTEMD_SYSCONFIG_DST_DEB
+fi
+
+# Check if system is using upstart
```

Comment:
There is no need to do this. Regardless of the init system used, you can just unconditionally install both the sysv service and the systemd unit -- as long as they have the same name, systemd will prefer its service, and sysv will just ignore it.